qemu/tests/drive_mirror_cluster_size.py

```diff
@@ -0,0 +1,105 @@
+import re
+import logging
+
+from avocado.utils import process
+
+from virttest import funcatexit
+from virttest import error_context
+from virttest import qemu_storage
+from virttest import data_dir
+
+from qemu.tests import drive_mirror
+
+
+def clean_trash_files(trash_files):
+    """
+    Clean the files created in the test, it will be
+    executed after the test finish.
+    :param trash_files: the file list need to be clean
+    """
+    while trash_files:
+        clean_cmd = "rm -f %s" % trash_files.pop()
+        process.system(clean_cmd, ignore_status=True)
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test block mirroring to different cluster size image.
+
+    1). create target image with different cluster size
+    2). boot vm, then mirror $source_image to $target_image
+    2). wait for mirroring job go into ready status
+    3). pause vm after vm in ready status
+    4). reopen $target_image file
+    5). compare $source image and $target_image file
+    6). resume vm
+    7). create target2 with default cluster size
+    8). mirror again to target2
+    9). Verify the cluster_size of target image correct
+    """
+
+    trash_files = []
+
+    def block_mirror(tag):
+        """
+        A block mirror process, including cluster_size verification.
+        It will be execute twice with different source and target.
+        :param tag: tag pass to DriveMirror object, source image
+        """
+        qemu_img = qemu_storage.QemuImg(
+            params, data_dir.get_data_dir(), tag)
+        mirror_test = drive_mirror.DriveMirror(test, params, env, tag)
+        source_image = mirror_test.get_image_file()
+        target_image = mirror_test.get_target_image()
+        mirror_test.create_files()
+        mirror_test.start()
+        mirror_test.action_when_steady()
+        mirror_test.vm.pause()
+        mirror_test.reopen()
+        device_id = mirror_test.vm.get_block({"file": target_image})
+        if device_id != mirror_test.device:
+            test.error("Mirrored image not being used by guest")
+            error_context.context("Compare fully mirrored images",
+                                  logging.info)
+        qemu_img.compare_images(source_image, target_image, force_share=True)
+        mirror_test.vm.resume()
+        mirror_test.verify_md5s()
```

Comment:
Can it be also defined in the cfg file as 'after_reopen = verify_md5s'?

Reply:
Same as above, then I must have `mirror_test.action_after_reopen()` in this line. That's same and unnecessary.